Dev/iwu/us vbs22 37 cambio contraseña nuevo

ichivancrCenfotec opened the pull request:
Changed Credentials

ichivancrCenfotec committed:
Cambio y recuperacion de contraseña. Adición de métodos para generar OTP y adición de atributo otp a user

ichivancrCenfotec committed:
Merge remote-tracking branch 'refs/remotes/origin/development' into dev/iwu/US-VBS22-37-Cambio_y_recuperacion_de_contraseña

# Conflicts:
#	demo/src/main/java/com/project/demo/logic/entity/user/User.java
#	demo/src/main/java/com/project/demo/logic/entity/user/UserRepository.java
#	demo/src/main/resources/application.properties

ichivancrCenfotec committed:
Update application.properties

Cambio de parámetros de aplication properties para variables de entorno

ichivancrCenfotec committed:
Update UserRepositoryTests.java

Resolución de conflictos

ichivancrCenfotec committed:
Merge branch 'dev/iwu/US-VBS22-37-Cambio_y_recuperacion_de_contraseña' into dev/iwu/US-VBS22-37-CAMBIO_CONTRASEÑA_NUEVO

ichivancrCenfotec committed:
Actaulización

Agregación de métodos para encontrar otp

ichivancrCenfotec committed:
Cambios

Cambios

ichivancrCenfotec committed:
Envio de correo

Envio de correo

ichivancrCenfotec committed:
Update

Update Test

ichivancrCenfotec committed:
Test

Adición de test

ichivancrCenfotec committed:
Changed Email Sender Credential

Changed Email Sender Credential

nefi-cenfo commented on demo/src/main/java/com/project/demo/entity/LoginResponse.java:
```diff
@@ -9,6 +9,15 @@ public class LoginResponse {
 
+    private Integer user_id;
+
+    public Integer getUser_id() {
```
Please consider to use camel case format.

nefi-cenfo commented on demo/src/main/java/com/project/demo/rest/AuthRestController.java:
```diff
@@ -84,4 +88,38 @@ public ResponseEntity<?> registerUser(@RequestBody User user) {
+
+    @PostMapping("/reset-password")
+    public ResponseEntity<?> resetPassword(@RequestBody User user) {
+        Optional<User> optionalUser = userRepository.findByEmail(user.getEmail());
```
Please try to use a service layer instead.

nefi-cenfo commented on demo/src/main/java/com/project/demo/rest/AuthRestController.java:
```diff
@@ -84,4 +88,38 @@ public ResponseEntity<?> registerUser(@RequestBody User user) {
+
+    @PostMapping("/validate-otp")
+    public ResponseEntity<?> validateOTP(@RequestBody User user) {
+        Optional<User> optionalUser = userRepository.findByOTP(user.getOtp());
```
Please try to use a service layer instead.

nefi-cenfo commented on demo/src/test/java/com/project/demo/repository/UserRepositoryTests.java:
```diff
@@ -125,60 +125,21 @@ public void infoSetup() {
-    }
-
-    @Test
-    public void testLoginSuccess() {
```
Why is this all code deleted?

nefi-cenfo commented on demo/src/test/java/com/project/demo/repository/UserRepositoryTests.java:
```diff
@@ -125,60 +125,21 @@ public void infoSetup() {
     }
 
     @Test
-    public void UserDelete() {
```
Watch out for deleting existing tests!!!

mur-tz committed:
Merge branch 'development' into dev/iwu/US-VBS22-37-CAMBIO_CONTRASEÑA_NUEVO

ichivancrCenfotec committed:
Update

Fixed problems with unit test. Authentication Controller functions implemented on service layer. Only validate OTP and reset-password

ichivancrCenfotec committed:
Bug fixed

Fixes test issues

ichivancrCenfotec committed:
Update

Test fixed again
